### PR TITLE
Task/11795/local cluster argo route

### PR DESCRIPTION
**Pull request by richiepiya**
**Related Issue:** https://github.com/open-cluster-management/backlog/issues/11795

Get Argo route information for local-cluster apps when Search is unavailable

### CODE REVIEW

**Review by fxiang1 — APPROVED** [review]
Looks good! Thanks!

File: src/v2/models/application.js [hunk 1]
```diff
@@ -857,13 +857,28 @@ export default class ApplicationModel extends GenericModel {
       }));
   }
 
+  async getLocalArgoRoute(args) {
+    const routes = await this.kubeConnector.getResources((ns) => `/apis/${args.apiVersion}/namespaces/${ns}/routes`, { namespaces: [args.namespace], kind: args.kind }).catch((err) => {
+      logger.error(err);
+      throw err;
+    });
+
+    return routes.filter((route) => _.get(route, 'metadata.labels["app.kubernetes.io/part-of"]') === 'argocd' && _.get(route, 'spec.host'))[0];
```
Comment (birsanv) [inline src/v2/models/application.js:866]: I don't see any check for corner cases Richie
- what happens if the routes is undefined ?
- route doesn't have that label
- you don't have to check for spec.host in here, just return the route
- what if there is not route with those props ? you get 0 elements

**Review by birsanv — CHANGES_REQUESTED** [review]
I don't see any test for corner cases Richie

const routes = await this.kubeConnector.getResources((ns) => `/apis/${args.apiVersion}/namespaces/${ns}/routes`, { namespaces: [args.namespace], kind: args.kind }).catch((err) => {
      logger.error(err);
      throw err;
    });

    return routes.filter((route) => _.get(route, 'metadata.labels["app.kubernetes.io/part-of"]') === 'argocd' && _.get(route, 'spec.host'))[0];

File: src/v2/models/application.js (hunk 1 above)
Comment (richiepiya) [reply src/v2/models/application.js:866]: Fixes added in new PR here: https://github.com/open-cluster-management/console-api/pull/318
